src/frontmatter/__tests__/fronmatter.spec.ts

```diff
@@ -11,17 +11,24 @@ const FIXTURES_DIR = join(__dirname, './fixtures');
 const tags = fs.readFileSync(join(FIXTURES_DIR, 'tags.md'), 'utf-8');
 
 describe('Frontmatter', () => {
+  it('should clone given data if dirty is set to false', () => {
```

Comment:
What is "dirty"?

Reply:
The naming might be a bit unfortunate, but the intention of the `dirty` flag is to indicate whether you want to have your AST mutated or not.
'dirty' (default) = mutable
'clean' (dirty == false) = immutable

Reply:
Maybe `mutate` would be more clear?